src/IndexedDbProvider.ts

```diff
@@ -276,9 +276,14 @@ export class IndexedDbProvider extends NoSqlProvider.DbProvider {
             return this._lockHelper.checkOpenTransaction(storeNamesArr, writeNeeded).then(() => {
                 const trans = this._db.transaction(intStoreNames, writeNeeded ? 'readwrite' : 'readonly');
                 const ourTrans = new IndexedDbTransaction(trans, this._schema, intStoreNames, this._fakeComplicatedKeys);
-                trans.oncomplete = () => {
+                const transcationUnlock = () => {
                     this._lockHelper.transactionComplete(storeNamesArr, writeNeeded);
                 };
+
+                trans.oncomplete = transcationUnlock;
```

Comment:
I think it's good to have it this way rather than have transaction logic broken in case of failure